lib/git_reflow.rb

```diff
@@ -107,53 +108,49 @@ def deliver(options = {})
-          puts "Merging pull request ##{existing_pull_request.number}: '#{existing_pull_request.title}', from '#{existing_pull_request.feature_branch_name}' into '#{existing_pull_request.base_branch_name}'"
-
-          update_destination(base_branch)
-          merge_feature_branch(feature_branch,
-                               :destination_branch  => base_branch,
-                               :pull_request_number => existing_pull_request.number,
-                               :lgtm_authors        => existing_pull_request.approvals,
-                               :message             => commit_message)
-          committed = run_command_with_label 'git commit', with_system: true
-
-          if committed
-            say "Merge complete!", :success
-
-            # check if user always wants to push and cleanup, otherwise ask
-            always_deploy_and_cleanup = GitReflow::Config.get('reflow.always-deploy-and-cleanup') == "true"
-            deploy_and_cleanup = always_deploy_and_cleanup || (ask "Would you like to push this branch to your remote repo and cleanup your feature branch? ") =~ /^y/i
-
-            if deploy_and_cleanup
-              run_command_with_label "git push origin #{base_branch}"
-              run_command_with_label "git push origin :#{feature_branch}"
-              run_command_with_label "git branch -D #{feature_branch}"
-              puts "Nice job buddy."
-            else
-              puts "Cleanup halted.  Local changes were not pushed to remote repo.".colorize(:red)
-              puts "To reset and go back to your branch run \`git reset --hard origin/#{base_branch} && git checkout #{feature_branch}\`"
-            end
+
+          # displays the current status and prompts user for confirmation
+          self.status base_branch
+          always_deliver = GitReflow::Config.get('reflow.always-deliver') == "true"
+          deliver = always_deliver || (ask "This is the current status of your Pull Request. Are you sure you want to deliver? ") =~ /^y/i
+          
+          if deliver
+            puts "Merging pull request ##{existing_pull_request.number}: '#{existing_pull_request.title}', from '#{existing_pull_request.feature_branch_name}' into '#{existing_pull_request.base_branch_name}'"
+
+            begin
+              existing_pull_request.merge!(destination_branch: base_branch, force: options['skip_lgtm'])
+
+              say "Pull Request successfully merged.", :success
+
+              # check if user always wants to push and cleanup, otherwise ask
+              always_cleanup = GitReflow::Config.get('reflow.always-cleanup') == "true"
+              cleanup = always_cleanup || (ask "Would you like to cleanup your feature branch? ") =~ /^y/i
+
+              if cleanup
+                # Pulls merged changes from remote base_branch
+                run_command_with_label "git pull origin #{base_branch}"
+                run_command_with_label "git push origin #{base_branch}"
```

Comment:
I don't think we want to push again here, since the `existing_pull_request.merge!` should have already pushed in the fallback/bitbucket scenario, and the merge happens on the remote in the Github API scenario.